src/batou_ext/config.py

```diff
@@ -123,36 +123,61 @@ class RegexPatch(batou.component.Component):
             pattern=r'^foo=\d+(\w+)',
             replacement=r'foo=27\1')
 
+    With adding the 'source' parameter, an external file is used for
+    being patched. In this case you will need to ensure with tools from
+    batou.lib.file the correct permissions etc. of the new file.
     """
 
     path = None
     namevar = 'path'
+
+    source = None
     pattern = None
     replacement = None
 
     def configure(self):
         self.pattern = re.compile(self.pattern, re.MULTILINE)
 
+        if not self.path:
+            raise ValueError("Path must be set")
```

Comment:
maybe change that to 

    raise ValueError("`path` must be set")

 if you're at it anyway.